rules/consistent-function-scoping.js

```diff
@@ -60,6 +60,13 @@ function checkReferences(scope, parent, scopeManager) {
 		);
 }
 
+const isReactHooks = scope =>
+	scope.block &&
+	scope.block.parent &&
+	scope.block.parent.callee &&
+	scope.block.parent.callee.type === 'Identifier' &&
+	scope.block.parent.callee.name === 'useEffect';
```

Comment:
There are more built-in hooks: https://reactjs.org/docs/hooks-reference.html

Reply:
Added https://github.com/sindresorhus/eslint-plugin-unicorn/pull/588/commits/4a4f06efd7e67ef779017cbf4b0c8cc8bd3cae66

Feel I don't need remove those hooks doesn't accept function parameter.